### ISSUE TITLE
Add dim option to EigenAnalysis to reduce the number of dimensions

### ISSUE THREAD

**Issue by juliohm** [JuliaML/TableTransforms.jl#17]
Currently we retain all scores. It would be nice to have an option to retain only first `dim` elements in the basis.

**Comment by juliohm** [comment]
@eliascarv I will assign you to the issue so that no one starts working on it in parallel 👍🏽
